planner/core/point_get_plan.go

```diff
@@ -1552,13 +1552,44 @@ func findInPairs(colName string, pairs []nameValuePair) int {
 	return -1
 }
 
+type subQueryChecker struct {
+	hasSubQuery bool
+}
+
+func (s *subQueryChecker) Enter(in ast.Node) (node ast.Node, skipChildren bool) {
+	switch in.(type) {
+	case *ast.SubqueryExpr:
+		s.hasSubQuery = true
+		return in, true
+	}
+
+	return in, false
+}
+
+func (s *subQueryChecker) Leave(in ast.Node) (ast.Node, bool) {
+	// Before we enter the sub-query, we should keep visiting its children.
+	return in, !s.hasSubQuery
+}
+
+func isExprHasSubQuery(expr ast.Node) bool {
+	checker := &subQueryChecker{}
```

Comment:
Better to create an object pool for it instead of creating a new one each time, otherwise it may cause some performance regression on TPCC.

Reply:
Added.